### ISSUE TITLE
[iOS] [Android] Crashes on 1.0.6

### ISSUE THREAD

**Issue by dericksa** [castle/castle-react-native#26]
Hello, we have launched an app containing the 1.0.6 version of the library, but we are receiving reports of crashes on Firebase:

For Android

```
Fatal Exception: java.lang.UnsupportedOperationException: Tried to obtain display from a Context not associated with  one. Only visual Contexts (such as Activity or one created with Context#createWindowContext) or ones created with Context#createDisplayContext are associated with displays. Other types of Contexts are typically related to background entities and may return an arbitrary display.
       at android.app.ContextImpl.getDisplay(ContextImpl.java:2602)
       at android.content.ContextWrapper.getDisplay(ContextWrapper.java:1022)
       at io.castle.android.highwind.Highwind.token(SourceFile:20)
       at io.castle.android.Castle.id(Castle.java:95)
       at io.castle.android.Castle.clientId(Castle.java:298)
       at io.castle.android.api.model.Context.init(Context.java:25)
       at io.castle.android.api.model.Context.create(Context.java:30)
       at io.castle.android.Castle.createContext(Castle.java:458)
       at io.castle.android.api.model.Event.<init>(Event.java:41)
       at io.castle.android.Castle.track(Castle.java:167)
       at io.castle.android.Castle.registerLifeCycleCallbacks(Castle.java:79)
       at io.castle.android.Castle.configure(Castle.java:121)
       at com.reactnativecastle.CastleModule.configureWithPublishableKey(CastleModule.kt:71)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:151)
       at com.facebook.react.bridge.queue.NativeRunnable.run(NativeRunnable.java)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
       at android.os.Looper.loop(Looper.java:246)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:226)
       at java.lang.Thread.run(Thread.java:923)
```

For iOS:

```
Fatal Exception: NSInternalInconsistencyException
0  CoreFoundation                 0x1a6c719d8 __exceptionPreprocess
1  libobjc.A.dylib                0x1baff4b54 objc_exception_throw
2  CoreFoundation                 0x1a6b8050c -[CFPrefsSearchListSource addManagedSourceForIdentifier:user:]
3  Foundation                     0x1a7e7a878 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:]
4  FrontBoardServices             0x1b5fe4614 -[FBSSerialQueue assertBarrierOnQueue]
5  FrontBoardServices             0x1b5f967f4 -[FBSScene updateClientSettings:withTransitionContext:]
6  FrontBoardServices             0x1b5f96a40 -[FBSScene updateClientSettingsWithTransitionBlock:]
7  FrontBoardServices             0x1b5f96954 -[FBSScene updateClientSettingsWithBlock:]
8  UIKitCore                      0x1a8ff74d8 -[FBSScene(UIApp) updateUIClientSettingsWithBlock:]
9  UIKitCore                      0x1a94f109c -[UIApplication setProximityEventsEnabled:]
10 UIKitCore                      0x1a90fa388 -[UIDevice setProximityMonitoringEnabled:]
11 Highwind                       0x104f34c08 (Ausente)
12 Highwind                       0x104f3583c (Ausente)
13 Highwind                       0x104f399d0 (Ausente)
14 Highwind                       0x104f33310 (Ausente)
15 Highwind                       0x104f32ff8 (Ausente)
16 MadeiraMadeira                 0x10479199c +[Castle configure:] + 92 (Castle.m:92)
17 MadeiraMadeira                 0x104791b3c +[Castle configureWithPublishableKey:] + 118 (Castle.m:118)
18 MadeiraMadeira                 0x1049e1310 -[RCTCastle configureWithPublishableKey:::] + 84 (RCTCastle.m:84)
19 CoreFoundation                 0x1a6c76130 __invoking___
20 CoreFoundation                 0x1a6b591d0 -[NSInvocation invoke]
21 CoreFoundation                 0x1a6b59730 -[NSInvocation invokeWithTarget:]
22 React                          0x1059b8868 (Ausente)
23 React                          0x1059baa0c (Ausente)
24 React                          0x1059ba770 (Ausente)
25 libdispatch.dylib              0x1a68a9298 _dispatch_call_block_and_release
26 libdispatch.dylib              0x1a68aa280 _dispatch_client_callout
27 libdispatch.dylib              0x1a68864f0 _dispatch_lane_serial_drain$VARIANT$armv81
28 libdispatch.dylib              0x1a6886fdc _dispatch_lane_invoke$VARIANT$armv81
29 libdispatch.dylib              0x1a6890800 _dispatch_workloop_worker_thread
30 libsystem_pthread.dylib        0x1ef4955a4 _pthread_wqthread
31 libsystem_pthread.dylib        0x1ef498874 start_wqthread
```

**Comment by aalekz** [comment]
Hey @dericksa!

Thanks for reporting these issues. On Android it looks like it is the same issue as this [Crash after upgrading to 2.0.x](https://github.com/castle/castle-android/issues/40). It has been resolved in version 2.0.2 of the Android SDK and we'll be updating the RN library to use the updated version in the next couple of days.

In regards to iOS it seem to be an issue with [Castle configure] not being called from the main thread in the native module, I've opened a [pull request with a fix](https://github.com/castle/castle-react-native/pull/28) for the issue. Once approved it will be included in the upcoming release.
